DBStartupScript/Daria/13-33 21-05-2021 log.log

```diff
@@ -0,0 +1,134 @@
+>> TerminatingError(): "Invalid configuration: databse size could not be greater than 1000 Gb"
+Invalid configuration: databse size could not be greater than 1000 Gb
+
+[91mException: [0mC:\Users\Daria\Desktop\DBStartupScript\Utils.ps1:31
+[96mLine |
+[96m  31 | [0m         [96mthrow "Invalid configuration: databse size could not be great[0m …
+[96m     | [91m         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+[91m[96m     | [91mInvalid configuration: databse size could not be greater than 1000 Gb
+[0m
+PS C:\Users\Daria\Desktop\DBStartupScript> ########################################################################
+PS C:\Users\Daria\Desktop\DBStartupScript> #removing "" to get [int] type for 50 instead of [string]
+PS C:\Users\Daria\Desktop\DBStartupScript> ########################################################################
+PS C:\Users\Daria\Desktop\DBStartupScript> TerminatingError(Get-Command): "Cannot validate argument on parameter 'Name'. The argument is null, empty, or an element of the argument collection contains a null value. Supply a collection that does not contain any null values and then try the command again."
+PS C:\Users\Daria\Desktop\DBStartupScript> . .\StartDatabase.ps1
+PS C:\Users\Daria\Desktop\DBStartupScript> ########################################################################
+PS C:\Users\Daria\Desktop\DBStartupScript> <#the result cannot be accepted: ** DB Server **
+
+Server started with following arguments:
+Database name:          TestDatabase
+Connection string:      \Test\DBS\01
+Database size:          50 Gb
+
+Tables:
+        System.Object[]#>
+PS C:\Users\Daria\Desktop\DBStartupScript> #"Tables" content is not provided. Fixing \Utils.ps in order to call each element of the massive
+PS C:\Users\Daria\Desktop\DBStartupScript> #########################################################################
+PS C:\Users\Daria\Desktop\DBStartupScript> TerminatingError(Get-Command): "Cannot validate argument on parameter 'Name'. The argument is null, empty, or an element of the argument collection contains a null value. Supply a collection that does not contain any null values and then try the command again."
+PS C:\Users\Daria\Desktop\DBStartupScript> . .\StartDatabase.ps1
+PS C:\Users\Daria\Desktop\DBStartupScript> #########################################################################
+PS C:\Users\Daria\Desktop\DBStartupScript> #the result is OK
+PS C:\Users\Daria\Desktop\DBStartupScript> <#** DB Server **
+
+Server started with following arguments:
+Database name:          TestDatabase
+Connection string:      \Test\DBS\01
+Database size:          50 Gb
+
+Tables:
+        Clients
+        Orders
+        Expenses
+        Deliveries#>
+PS C:\Users\Daria\Desktop\DBStartupScript> Stop-Transcript
+**********************
+PowerShell transcript end
+End time: 20210521134605
+**********************
```

Comment:
Logging be implemented in many ways but in it's essence it's writing down what is going on at the important points of script's execution. For example, you may update `Start-DatabaseServer` function as follows:
```
function Start-DatabaseServer()
{
    param(
        [Parameter(Mandatory=$true)]
        [PSCustomObject] $Configuration
    )

    Write-Host "Starting database server..."

    $argumentList = @($Configuration.DatabaseName, $Configuration.DatabaseConnectionString, $Configuration.Size, $Configuration.Tables -Split ",")
    Start-Process -FilePath .\Bin\DBServer.exe -ArgumentList ($argumentList)
} 
```
Note the `Write-Host "Starting database server..."` line - it will print out the "Starting..." line to the console when that function is called, so when you look at the console, you'll see when this function has been executed. If, for example, you add similar lines to every function, you'll be able to see all the steps being executed in the console. 🤓

You need something similar to this but it has to write to a file instead of a console. Consider writing a function similar to `Write-Host` which will write to a file of your choosing. 😉